gremlin-driver/src/main/java/example/Example.java

```diff
@@ -0,0 +1,163 @@
+        Cluster cluster = Cluster.build().
+            channelizer("org.apache.tinkerpop.gremlin.driver.Channelizer$HttpChannelizer").
+            keepAliveInterval(180000).
+            maxConnectionPoolSize(8).
+            path("/gremlin").
+            port(8192).
+            serializer(new GraphBinaryMessageSerializerV1()).
+            create();
+        g = traversal().withRemote(DriverRemoteConnection.using(cluster, "g"));
+
+        // Connecting and specifying a serializer
+        IoRegistry registry = new FakeIoRegistry(); // an IoRegistry instance exposed by a specific graph provider
+        TypeSerializerRegistry typeSerializerRegistry = TypeSerializerRegistry.build().addRegistry(registry).create();
+        MessageSerializer serializer = new GraphBinaryMessageSerializerV1(typeSerializerRegistry);
+        cluster = Cluster.build().
+            serializer(serializer).
+            create();
+        Client client = cluster.connect();
+        g = traversal().withRemote(DriverRemoteConnection.using(client, "g"));
+
+        // Cleanup
+        g.close();
+    }
+    public static class FakeIoRegistry extends AbstractIoRegistry {
+    }
+
+    private static void basicGremlinExample() {
+        Graph graph = TinkerGraph.open();
+        GraphTraversalSource g = traversal().withEmbedded(graph);
+
+        // Basic Gremlin: adding and retrieving data
+        Vertex v1 = g.addV("person").property("name","marko").next();
+        Vertex v2 = g.addV("person").property("name","stephen").next();
+        Vertex v3 = g.addV("person").property("name","vadas").next();
+
+        // Be sure to use a terminating step like next() or iterate() so that the traversal "executes"
+        // Iterate() does not return any data and is used to just generate side-effects (i.e. write data to the database)
+        g.V(v1).addE("knows").to(v2).property("weight",0.75).iterate();
+        g.V(v1).addE("knows").to(v3).property("weight",0.75).iterate();
+
+        // Retrieve the data from the "marko" vertex
+        Object marko = g.V().has("person","name","marko").values("name").next();
+        System.out.println("name: " + marko);
+
+        // Find the "marko" vertex and then traverse to the people he "knows" and return their data
+        List<Object> peopleMarkoKnows = g.V().has("person","name","marko").out("knows").values("name").toList();
+        for (Object person : peopleMarkoKnows) {
+            System.out.println("marko knows " + person);
+        }
+    }
+
+    private static void modernTraversalExample() {
+        /*
+        This example requires the Modern toy graph to be preloaded upon launching the Gremlin server.
+        For details, see https://tinkerpop.apache.org/docs/current/reference/#gremlin-server-docker-image and use
+        conf/gremlin-server-modern.yaml.
+        */
+        Graph modern = TinkerFactory.createModern();
+        GraphTraversalSource g = traversal().withEmbedded(modern);
```

Comment:
this is embedded traversal, so no server required. Need to change comment or connection setup

Reply:
Edited comment to provide context on usage of TinkerFactory instead